<launch>

  <!-- parameters-->
  <param name="/detect_ball_down_orange/h_min_orange" value="0" />
  <param name="/detect_ball_down_orange/h_max_orange" value="55" />
  <param name="/detect_ball_down_orange/s_min_orange" value="0" />
  <param name="/detect_ball_down_orange/s_max_orange" value="255" />
  <param name="/detect_ball_down_orange/v_min_orange" value="195" />
  <param name="/detect_ball_down_orange/v_max_orange" value="255" />

  <node name="detect_ball_down_orange" pkg="coconuts_odroid" type="detect_ball_down_orange" />

  <!-- parameters-->
  <param name="/detect_ball_down_green/h_min_green" value="21" />
  <param name="/detect_ball_down_green/h_max_green" value="60" />
  <param name="/detect_ball_down_green/s_min_green" value="0" />
  <param name="/detect_ball_down_green/s_max_green" value="255" />
  <param name="/detect_ball_down_green/v_min_green" value="195" />
  <param name="/detect_ball_down_green/v_max_green" value="255" />

  <node name="detect_ball_down_green" pkg="coconuts_odroid" type="detect_ball_down_green" />

  <!-- stereo image proc - down camera  --> 
  <!-- <node ns="camera_down" name= "image_proc" pkg="image_proc" type="image_proc"/>  -->

  <!-- downward camera uses usb_cam, this one is not on the USB bus -->
  <node name= "camera_down" pkg="usb_cam" type="usb_cam_node" >
    <param name="video_device" type="string" value="/dev/video2"/>
    <param name="framerate" type="int" value="5"/>
    <param name="camera_frame_id" type="string" value="camera_down" /> 
    <param name="camera_info_url" type="string" value="file://$(find coconuts_odroid)/src/camera_down_calibration.yaml"/>
  </node> 

  <!-- locate ball publishes ball transform in 3d space -->
  <node name="locate_ball_down" pkg="coconuts_odroid" type="locate_ball_down" output="screen" />


  </launch>
